test/functional/fixtures/regression/gh-851/test.js

```diff
@@ -0,0 +1,5 @@
+describe('[Regression](GH-851)', function () {
+    it('Should call click for common parent element', function () {
```

Comment:
You've added code with a lot of conditions. I suppose we should have several tests for them. And at least one test for dblclick changes.